### DISCUSSION TITLE
Error on OS/2

### DISCUSSION THREAD

**Discussion by Tellie** [pawelsalawa/sqlitestudio#5425] [General]
Hi,
I try to compile this under good old OS/2.
Using Qt5.15.2 and gcc9.2.0

When building it i get those errors:
g++ -c -Zomf -march=i686 -pedantic -O2 -g -march=pentium4 -msse2 -std=gnu++1z -Wall -Wextra -DQT_NO_IPV6 -DPROJECT_MODULE_NAME=sqlitestudiocli -DQT_NO_DEBUG -DQT_CORE_LIB -I../../sqlitestudiocli -I. -I../../coreSQLiteStudio -I/@unixroot/usr/include/qt5 -I/@unixroot/usr/include/qt5/QtCore -I../../../output/build/sqlitestudiocli -I/@unixroot/usr/lib/qt5/mkspecs/os2-g++ -o ../../../output/build/sqlitestudiocli/clicommandsql.obj ../../sqlitestudiocli/commands/clicommandsql.cpp
../../sqlitestudiocli/commands/clicommandsql.cpp: In member function 'void CliCommandSql::printResultsFixed(QueryExecutor*, SqlQueryPtr)':
../../sqlitestudiocli/commands/clicommandsql.cpp:134:40: error: 'termCols' was not declared in this scope
  134 |     if ((resultColumnsCount * 2 - 1) > termCols)
      |                                        ^~~~~~~~
../../sqlitestudiocli/commands/clicommandsql.cpp:144:17: error: 'baseColWidth' was not declared in this scope
  144 |         width = baseColWidth;
      |                 ^~~~~~~~~~~~
../../sqlitestudiocli/commands/clicommandsql.cpp:146:23: error: 'termCols' was not declared in this scope
  146 |             width += (termCols - resultColumnsCount * (baseColWidth + 1) + 1);
      |                       ^~~~~~~~
../../sqlitestudiocli/commands/clicommandsql.cpp: In member function 'void CliCommandSql::printResultsColumns(QueryExecutor*, SqlQueryPtr)':
../../sqlitestudiocli/commands/clicommandsql.cpp:174:9: error: 'resultColumnsCount' was not declared in this scope; did you mean 'resultColumns'?
  174 |     if (resultColumnsCount == 0)
      |         ^~~~~~~~~~~~~~~~~~
      |         resultColumns
../../sqlitestudiocli/commands/clicommandsql.cpp:178:10: error: 'resultColumnsCount' was not declared in this scope; did you mean 'resultColumns'?
  178 |     if ((resultColumnsCount * 2 - 1) > termCols)
      |          ^~~~~~~~~~~~~~~~~~
      |          resultColumns
../../sqlitestudiocli/commands/clicommandsql.cpp:178:40: error: 'termCols' was not declared in this scope
  178 |     if ((resultColumnsCount * 2 - 1) > termCols)
      |                                        ^~~~~~~~
../../sqlitestudiocli/commands/clicommandsql.cpp:201:29: error: 'resultColumnsCount' was not declared in this scope; did you mean 'resultColumns'?
  201 |         for (int i = 0; i < resultColumnsCount; i++)
      |                             ^~~~~~~~~~~~~~~~~~
      |                             resultColumns
../../sqlitestudiocli/commands/clicommandsql.cpp:213:20: error: 'resultColumnsCount' was not declared in this scope; did you mean 'resultColumns'?
  213 |     totalWidth += (resultColumnsCount - 1); // column separators
      |                    ^~~~~~~~~~~~~~~~~~
      |                    resultColumns
../../sqlitestudiocli/commands/clicommandsql.cpp:216:22: error: 'termCols' was not declared in this scope
  216 |     if (totalWidth < termCols)
      |                      ^~~~~~~~
../../sqlitestudiocli/commands/clicommandsql.cpp: In member function 'void CliCommandSql::printResultsRowByRow(QueryExecutor*, SqlQueryPtr)':
../../sqlitestudiocli/commands/clicommandsql.cpp:270:38: error: 'termWidth' was not declared in this scope
  270 |         qOut << center(rowCntString, termWidth - 1, '-') << "\n";
      |                                      ^~~~~~~~~
make[1]: *** [Makefile:1404: ../../../output/build/sqlitestudiocli/clicommandsql.obj] Error 1
 

Any idea what i miss in here??

With kind regards, Tellie

**Comment by tuffnatty** [comment]
@TeLLie Hard to say without setting up a build environment, but your line numbers are off-by-one in comparison to Git HEAD and 3.4.2-3.4.4 releases.
